plugins/modules/packaging/os/homebrew_cask.py

```diff
@@ -454,18 +453,45 @@ def _current_cask_is_installed(self):
             self.message = 'Invalid cask: {0}.'.format(self.current_cask)
             raise HomebrewCaskException(self.message)
 
-        cmd = [
-            "{brew_path}".format(brew_path=self.brew_path),
-            "cask",
-            "list",
-            self.current_cask
-        ]
+        if self._brew_cask_command_is_deprecated():
+            base_opts = [
+                "{brew_path}".format(brew_path=self.brew_path),
+                "list",
+                "--cask",
+            ]
+        else:
+            base_opts = [
+                "{brew_path}".format(brew_path=self.brew_path),
+                "cask",
+                "list",
+            ]
+
+        cmd = base_opts + [self.current_cask]
         rc, out, err = self.module.run_command(cmd)
 
         if rc == 0:
             return True
         else:
             return False
+
+    def _get_brew_version(self):
+        cmd = [self.brew_path, '--version']
+
+        process = subprocess.run(cmd, stdout=subprocess.PIPE)
+
+        process_output = process.stdout.decode()
+
+        # get version string from first line of "brew --version" output
+        version = process_output.split('\n')[0].split(' ')[1]
+
+        return version
```

Comment:
How about caching the value of `version`, so the module doesn't call `brew --version` possibly multiple times per module execution?

Reply:
good point! i will make the change